File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[]{0,0};
        // key = num, val = index
        Map<Integer, Integer> m = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];

            if (m.containsKey(difference)) {
                result[0] = m.get(difference);
                result[1] = i;
                return result;
            }

            m.put(nums[i], i);
        }

        return result;
    }
}

File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        res = 0
        new = []
        for num in nums:
            if num != val: 
                res += 1
                new.append(num)


        print(nums, new)

        for i in range(len(new)):
            nums[i] = new[i]

        return res
                